### PR TITLE
Pushdown filters timeseries

### PR DESCRIPTION
**Pull request by hakontro**
Pushdown filters on assetId for Time Series.

### CODE REVIEW

File: src/main/scala/com/cognite/spark/datasource/DataPointsRddV1.scala [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  import CdpConnector.sttpBackend
+  @transient lazy val client =
+    new GenericClient[IO, Nothing](Constants.SparkDatasourceVersion)
+
+  override def getPartitions: Array[Partition] = {
+    val numberofIOs = getIOs(client).length
+    0.until(numberofIOs).toArray.map(CdfPartition)
+  }
+  override def compute(_split: Partition, context: TaskContext): Iterator[Row] = {
+
```
Comment (wjoel) [inline src/main/scala/com/cognite/spark/datasource/DataPointsRddV1.scala]: So many newlines! I'd just do `getIOs(client)(split.index)`

File: src/main/scala/com/cognite/spark/datasource/TimeSeriesRelation.scala [hunk 2]
```diff
@@ -88,10 +99,24 @@ class TimeSeriesRelation(config: RelationConfig)(val sqlContext: SQLContext)
...
-      config.limit.map(client.timeSeries.listWithLimit(_)).getOrElse(client.timeSeries.list)
-    )
+      numPartitions: StatusCode): Seq[fs2.Stream[IO, TimeSeries]] = {
+
+    val fieldNames = Array("assetId")
+    val pushdownFilterExpression = toPushdownFilterExpression(filters)
+    val shouldGetAllRows = shouldGetAll(pushdownFilterExpression, fieldNames)
+    val filtersAsMaps = pushdownToParameters(pushdownFilterExpression)
+    val assetIds = filtersAsMaps.flatMap(m => m.get("assetId").map(_.toLong))
+    if (assetIds.isEmpty || shouldGetAllRows) {
```
Comment (wjoel) [inline src/main/scala/com/cognite/spark/datasource/TimeSeriesRelation.scala]: Should use `limit` not `config.limit`, and `client.timeSeries.filter` if it's not present:
```scala
    if (filtersAsMaps.isEmpty || shouldGetAllRows) {
      Seq(
        limit
          .map(client.timeSeries.listWithLimit)
          .getOrElse(client.timeSeries.list)
      )
    } else {
      val filter = TimeSeriesFilter(assetIds = assetIds)
      Seq(
        limit
          .map(l => client.timeSeries.filterWithLimit(filter, l))
          .getOrElse(client.timeSeries.filter(filter))
      )
    }
```

File: src/test/scala/com/cognite/spark/datasource/TimeSeriesRelationTest.scala [hunk 3]
```diff
@@ -26,6 +26,21 @@ class TimeSeriesRelationTest extends FlatSpec with Matchers with SparkTest {
...
+  it should "handle pushdown filters on assetIds" taggedAs WriteTest in {
+    val metricsPrefix = "pushdown.filters.assetIds"
+    val df = spark.read
+      .format("com.cognite.spark.datasource")
+      .option("apiKey", readApiKey.apiKey)
+      .option("type", "timeseries")
+      .option("collectMetrics", "true")
+      .option("metricsPrefix", metricsPrefix)
+      .load()
+      .where(s"assetId In(6191827428964450, 3424990723231138, 3047932288982463)")
```
Comment (wjoel) [inline src/test/scala/com/cognite/spark/datasource/TimeSeriesRelationTest.scala]: Add a test for `assetId in ()` and one where `assetId in (9999)` or other invalid id (should read 0 rows).

Comment (hakontro) [reply src/test/scala/com/cognite/spark/datasource/TimeSeriesRelationTest.scala]: Good tip! Turns out Spark doesn't like `in()`, so didn't add one for that.

**Pull request merged**